autoarray/plot/mat_wrap/mat_plot.py

```diff
@@ -110,31 +110,72 @@ def __init__(
+        self.figure = figure or wb.Figure(is_default=True)
+        self.axis = axis or wb.Axis(is_default=True)
 
-        self.cmap = cmap or wb.Cmap()
+        self.cmap = cmap or wb.Cmap(is_default=True)
 
-        self.colorbar = colorbar or wb.Colorbar()
-        self.colorbar_tickparams = colorbar_tickparams or wb.ColorbarTickParams()
+        self.colorbar = colorbar or wb.Colorbar(is_default=True)
+        self.colorbar_tickparams = colorbar_tickparams or wb.ColorbarTickParams(is_default=True)
 
-        self.tickparams = tickparams or wb.TickParams()
-        self.yticks = yticks or wb.YTicks()
-        self.xticks = xticks or wb.XTicks()
+        self.tickparams = tickparams or wb.TickParams(is_default=True)
+        self.yticks = yticks or wb.YTicks(is_default=True)
+        self.xticks = xticks or wb.XTicks(is_default=True)
 
-        self.title = title or wb.Title()
-        self.ylabel = ylabel or wb.YLabel()
-        self.xlabel = xlabel or wb.XLabel()
+        self.title = title or wb.Title(is_default=True)
+        self.ylabel = ylabel or wb.YLabel(is_default=True)
+        self.xlabel = xlabel or wb.XLabel(is_default=True)
 
-        self.text = text or wb.Text()
-        self.legend = legend or wb.Legend()
-        self.output = output or wb.Output()
+        self.text = text or wb.Text(is_default=True)
+        self.legend = legend or wb.Legend(is_default=True)
+        self.output = output or wb.Output(is_default=True)
 
         self.number_subplots = None
         self.subplot_shape = None
         self.subplot_index = None
 
+    def __add__(self, other):
+        """
+        Adds two `MatPlot` classes together.
+
+        A `MatPlot` class contains many of the `MatWrap` objects which customize matplotlib figures. One
+        may have a standard `MatPlot` object, which customizes many figures in the same way, for example:
+
+        mat_plot_2d_base = aplt.MatPlot2D(
+            yticks=aplt.YTicks(fontsize=18),
+            xticks=aplt.XTicks(fontsize=18),
+            ylabel=aplt.YLabel(label=""),
+            xlabel=aplt.XLabel(label=""),
+        )
+
+        However, one may require many unique `MatPlot` objects for a number of different figures, which all use
+        these settings. These can be created by creating the unique `MatPlot` objects and adding the above object
+        to each:
+
+        mat_plot_2d = aplt.MatPlot2D(
+            title=aplt.Title(label="Example Figure 1"),
+        )
+
+        mat_plot_2d = mat_plot_2d + mat_plot_2d_base
+
+        mat_plot_2d = aplt.MatPlot2D(
+            title=aplt.Title(label="Example Figure 2"),
+        )
+
+        mat_plot_2d = mat_plot_2d + mat_plot_2d_base
+        """
+
+        for attr, value in self.__dict__.items():
+
+            try:
+                if value.kwargs.get("is_default") is not True:
+                    other.__dict__[attr] = value
+            except AttributeError:
+                pass
+
+        return other

```

Comment:
This add function also mutates the original other class. Consider deep copying it at the start to avoid unexpected behaviour